addons/point_of_sale/models/account_journal.py

```diff
@@ -15,6 +15,25 @@ def _check_type(self):
         if methods:
             raise ValidationError(_("This journal is associated with a payment method. You cannot modify its type"))
 
+    def has_active_payments(self):
+        hanging_journal_entries = self.env['pos.payment'].search_count(
+        [
+            ('payment_method_id.id', 'in', self.pos_payment_method_ids.ids),
+            ('session_id.state', '=', 'opened')
+        ], limit=1)
+        return hanging_journal_entries
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_journal_except_with_active_payments(self):
+        for journal in self:
+            if journal.has_active_payments():
+                raise ValidationError(_("This journal is associated with a payments not finished in an active pos session"))
```

Comment:
```suggestion
                raise ValidationError(_("This journal is associated with payment method %s that is being used by order %s in the active pos session %s"))
```

You could retrieve one pos order which breaks the check in order to tell the user which record is breaking the check.

Also, if you could add the translation string into the .pot (not needed in the .po) it would be great. POTs are regenerated when adding features in master, but not in stable.